modules/dnn/include/opencv2/dnn/all_layers.hpp

```diff
@@ -1318,6 +1318,17 @@ CV__DNN_INLINE_NS_BEGIN
         static Ptr<CenterCropPadLayer> create(const LayerParams& params);
     };
 
+    class CV_EXPORTS DFTLayer : public Layer
+    {
+    public:
+        bool inverse;
+        bool onesided;
+        int axis_attr;
+        int dft_length;
```

Comment:
please, remove dft_length from attributes, because it's input, not an attribute. Compute dft_length only inside getMemoryShapes (if the second input is constant) and inside forward. Otherwise, the logic becomes quite fuzzy.